syz-manager/rpc.go

```diff
@@ -312,6 +346,61 @@ func (serv *RPCServer) NewInput(a *rpctype.NewInputArgs, r *int) error {
 	return nil
 }
 
+func parseRPCInput(rg *cover.ReportGenerator, modules []*host.KernelModule, inp rpctype.RPCInput) map[string][]uint32 {
+	offsets := make(map[string][]uint32)
+	for _, pc := range inp.Cover {
+		pc1 := rg.RestorePC(pc)
+		if len(modules) < 2 {
+			offsets[""] = append(offsets[""], pc)
+			continue
+		}
+		name, offset := findModuleOffset(pc1, modules)
+		offsets[name] = append(offsets[name], offset)
+	}
+	return offsets
+}
+
+func getNewBitmapFilter(rg *cover.ReportGenerator, pcs map[uint32]uint32,
+	modules1, modules2 []*host.KernelModule) map[uint32]uint32 {
+	if len(modules1) < 2 {
+		return pcs
+	}
+	pcs1 := make(map[uint32]uint32)
+	for pc := range pcs {
+		pc1 := rg.RestorePC(pc)
+		name, offset1 := findModuleOffset(pc1, modules1)
+		for _, module := range modules2 {
+			if name == module.Name {
+				pc2 := uint32(module.Addr + uint64(offset1))
+				pcs1[pc2] = 1
+				break
+			}
+		}
+	}
+	return pcs1
+}
+
+func findModuleOffset(pc uint64, modules []*host.KernelModule) (string, uint32) {
+	idx := sort.Search(len(modules), func(i int) bool {
+		return pc < modules[i].Addr
+	})
+	if idx == 0 && pc < modules[0].Addr {
+		log.Fatalf("pc (%v) < modules[0].Addr (%v), something wrong with syzkaller", pc, modules[0].Addr)
+	}
+	idx--
+	if modules[0].Name == "" {
+		if pc < modules[1].Addr {
+			return "", uint32(pc)
+		}
+	} else if modules[len(modules)-1].Name == "" {
```

Comment:
Can you please clarify what are these situations (the first entry of modules with empty `Name` and the last module with the empty `Name`)? How they are possible and how are they different?

Reply:
Agreed with Dmitry in previous PR that "" stands for vmlinux in linux.